### DISCUSSION TITLE
Anybody knows why some directories are not mounted for my v3

### DISCUSSION THREAD

**Discussion by hua-bing** [gtxaspec/wz_mini_hacks#670] [General]
Hopefully it is appropriate to ask here.

I have a v3 that won't boot and showing solid red light.  I see from the log that it's missing these mounting points:
tmpfs on /dev type tmpfs (rw,relatime)
devpts on /dev/pts type devpts (rw,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,relatime,mode=755)
media on /media type tmpfs (rw,relatime)
/dev/mmcblk0p1 on /lib/modules type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
/dev/mmcblk0p1 on /media/mmc type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)`

The v3_init.log shows that it is waiting for wifi vendor id.  Is this because a hardware failure or corrupted OS?  I tried to reflash the firmware, but it didn't help.

This is the mount info. from a healthy Wyze Cam v3 running wz_mini_hacks:
rootfs on / type rootfs (rw)
devtmpfs on /dev type devtmpfs (rw,relatime,size=46444k,nr_inodes=11611,mode=755)
proc on /proc type proc (rw,relatime)
sysfs on /sys type sysfs (rw,relatime)
/dev/mmcblk0p1 on /opt type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
/dev/mtdblock2 on / type squashfs (ro,relatime)
/dev/mmcblk0p1 on /bin/busybox type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
/dev/mtdblock6 on /configs type jffs2 (rw,relatime)
/dev/mmcblk0p1 on /etc/inittab type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
/dev/mmcblk0p1 on /etc/profile type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
tmpfs on /tmp type tmpfs (rw,relatime)
/dev/mtdblock3 on /system type squashfs (ro,relatime)
/dev/mmcblk0p1 on /system/bin/factorycheck type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
/dev/mmcblk0p1 on /etc/fstab type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
tmpfs on /opt/wz_mini/tmp type tmpfs (rw,relatime)
tmpfs on /etc/shadow type tmpfs (rw,relatime)
tmpfs on /dev type tmpfs (rw,relatime)
devpts on /dev/pts type devpts (rw,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,relatime,mode=755)
media on /media type tmpfs (rw,relatime)
/dev/mmcblk0p1 on /lib/modules type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
/dev/mmcblk0p1 on /media/mmc type vfat (rw,relatime,fmask=0000,dmask=0000,allow_utime=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)`

**Comment by archandanime** [comment]
About the Wi-Fi vendor ID: The Wyze Cam v3 needs to figure out what Wi-Fi module physically presents on the camera.

For all T20 cameras, the Wi-Fi module is RTL8189FTV. But on Wyze Cam v3(T31), the Wi-Fi module can be either `RTL8189FTV` or `ATBM6032x`(hence the 2 Wi-Fi module options on `wz_mini.conf`).

In order to load the correct Wi-Fi driver,wz_mini_hacks and stock firmware detect the Wi-Fi module by activating the Wi-Fi module via GPIO59 then read the Wi-Fi vendor ID returned by the module. But in your case the Wi-Fi module doesn't return anything, hence the `waiting for wifi vendir id` error.

If stock firmware doesn't work, I believe it is hardware error.

**Reply by hua-bing** [reply]
Thanks @archandanime.  From app_init.sh, it stops at:
`while [ ! -f /sys/bus/mmc/devices/mmc1\:0001/mmc1\:0001\:1/vendor ]; do`
`    sleep 0.1`
`done`

Not sure what caused the hardware failure.  It just died suddenly.  Quite unusual for me.  Maybe I better stop playing with it.

**Reply by archandanime** [reply]
If you feel adventurous and have serial connection to the camera, enable `DEBUG_INITRAMFS_ENABLED="true"` on `wz_mini.conf` and try to read the Wi-Fi vendor ID manually by running these commands(you can copy-paste the all commands at once):
```
wifi_gpio_pin="59"
	
if [ ! -d /sys/class/gpio/gpio$wifi_gpio_pin ]; then
	echo $wifi_gpio_pin > /sys/class/gpio/export
	echo out > /sys/class/gpio/gpio$wifi_gpio_pin/direction
	echo 1 > /sys/class/gpio/gpio$wifi_gpio_pin/value
fi
	
echo INSERT > /sys/devices/platform/jzmmc_v1.2.1/present
sleep 3

cat /sys/bus/mmc/devices/mmc1:0001/mmc1:0001:1/vendor
```
If the last `cat` command returns `0x024c` or `0x007a`, your Wi-Fi module is okay but if it still returns nothing, I guess you have a dead Wi-Fi module then.
You might want to do it several times to make sure.

**Reply by hua-bing** [reply]
Thanks @archandanime.  Looks it'll be a bit involved.  At least I don't yet have anything to make a serial connection.  I'll revisit if I find more time :)

**Comment by hua-bing** [comment]
Closing for now.
